app/federation-v2/server/matrix-client/user.ts

```diff
@@ -79,17 +84,25 @@ export const invite = async (inviterId: string, roomId: string, invitedId: strin
 		console.log(`[${inviterId}-${invitedId}-${roomId}] Invited user created as ${invitedUserMatrixId}...`);
 	}
 
-	console.log(`[${inviterId}-${invitedId}-${roomId}] Inviting the user...`);
-
-	// Invite the user
-	await matrixBridge.getIntent(inviterUser.mui).invite(matrixRoomId, invitedUserMatrixId);
+	console.log(`[${inviterId}-${invitedId}-${roomId}] Inviting the user to the room...`);
 
-	// Auto-join if the user is Rocket.Chat controlled
+	// Invite && Auto-join if the user is Rocket.Chat controlled
 	if (!invitedUserIsRemote) {
+
+		// Invite the user to the room
+		await matrixBridge.getIntent(inviterUser.mui).invite(matrixRoomId, invitedUserMatrixId);
+
 		console.log(`[${inviterId}-${invitedId}-${roomId}] Auto-join room...`);
 
 		await matrixBridge.getIntent(invitedUserMatrixId).join(matrixRoomId);
+	} else {
+		// Invite the user to the room but don't wait as this is dependent on the user accepting the invite because we don't control this user
+		matrixBridge.getIntent(inviterUser.mui).invite(matrixRoomId, invitedUserMatrixId);
```

Comment:
might be more elegant way.  But when inviting a remote user it would hang forever and not proceed.  This just fires and forgets as it won't get resolved fast enough since human has to accept